contracts/products/SoteriaCoverageProduct.sol

```diff
@@ -0,0 +1,681 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "../utils/Governable.sol";
+import "../interfaces/utils/IRegistry.sol";
+import "../interfaces/risk/IRiskManager.sol";
+import "../interfaces/products/ISoteriaCoverageProduct.sol";
+
+/**
+ * @title SoteriaCoverageProduct
+ * @author solace.fi
+ * @notice The smart contract implementation of **SoteriaCoverageProduct**.
+ */
+contract SoteriaCoverageProduct is ISoteriaCoverageProduct, ERC721, EIP712, ReentrancyGuard, Governable {
+    using Address for address;
+
+    /***************************************
+    GLOBAL VARIABLES
+    ***************************************/
+
+    /// @notice Registry contract.
+    IRegistry internal _registry;
+
+    /// @notice Cannot buy new policies while paused. (Default is False)
+    bool internal _paused;
+
+    /***************************************
+    BOOK-KEEPING VARIABLES
+    ***************************************/
+
+    /// @notice The total policy count.
+    uint256 internal _totalPolicyCount;
+
+    /**
+     * @notice The maximum rate charged per second per wei of coverLimit.
+     * @dev Because Solidity cannot store fractions in a single variable, need two variables: one for numerator and one for divisor
+     * @dev We also need to be careful to perform multiplication before division, as Solidity rounds down to 0
+     * @dev For testing assume _maxRate reflects 10% of coverLimit annually = 1/315360000
+     */
+    uint256 internal _maxRateNum;
```

Comment:
Can we set following state variables in constructor? Or Can we set values when they are defined?
    uint256 internal _maxRateNum;
    uint256 internal _maxRateDenom;
    uint256 internal _chargeCycle;

Reply:
probably easier to set inside the constructor, right?

Reply:
Yes, let's define default values for those variables.

Reply:
ok so i'll set the following as default values in the constructor

uint256 internal _maxRateNum = 1; // Gives us annual max rate of 10% of coverLimit
uint256 internal _maxRateDenom = 315360000; // Gives us annual max rate of 10% of coverLimit
uint256 internal _chargeCycle = 604800; // One week

Reply:
Don't set those in the contract, they should be inputs in the constructor.